tests/nested/extra-snaps-assertions/task.yaml

```diff
@@ -0,0 +1,73 @@
+summary: create ubuntu-core image and execute the suite in a nested qemu instance
+
+systems: [ubuntu-16.04-64]
+
+prepare: |
+    . $TESTSLIB/prepare.sh
+    prepare_classic
+    prepare_each_classic
+
+    # determine arch related vars
+    case "$NESTED_ARCH" in
+    amd64)
+        QEMU="$(which qemu-system-x86_64)"
+        ;;
+    i386)
+        QEMU="$(which qemu-system-i386)"
+        ;;
+    *)
+        echo "unsupported architecture"
+        exit 1
+        ;;
+    esac
+
+    # create ubuntu-core image
+    mkdir -p /tmp/work-dir
+    snap install --devmode --beta ubuntu-image
```

Comment:
Installing this ubuntu-image snap will bundle its own /usr/bin/snap - so this test (as is) will not work, will it? We can probably tweak it by `mount -o bind /usr/bin/snap /snap/ubuntu-image/current/bin/snap` ?